server/pages/index.js

```diff
@@ -474,13 +480,46 @@ function render404( request, response ) {
 	response.status( 404 ).send( renderJsx( '404', ctx ) );
 }
 
+function handleLocaleSubdomains( req, res, next ) {
+	const langRouteParams = config( 'magnificent_non_en_locales' ).join( '|' );
+	const hostnameGroup = `\\.${ config( 'hostname' ).replace( /\./g, '\\.' ) }`;
+	const langSubDomainRegEx = new RegExp( `^(${ langRouteParams })(${ hostnameGroup })` );
+	const langSubDomainMatch = req.hostname.match( langSubDomainRegEx );
+
+	// If the subdomain is a valid route.
+	if ( langSubDomainMatch && langSubDomainMatch[ 1 ] ) {
+		// We get the language again for the RTL information.
+		const language = getLanguage( langSubDomainMatch[ 1 ] );
+
+		// We want to switch locales only on the themes page
+		// and in a logged-out state.
+		if ( language && ! req.cookies.wordpress_logged_in && startsWith( req.path, '/themes' ) ) {
```

Comment:
Individual 'Theme Sheets' are at e.g. `/theme/twentyseventeen`. So we should either whitelist `/theme` too, or maybe drop the `startsWith()` condition altogether, as discussed here https://github.com/Automattic/wp-calypso/pull/25296#issuecomment-395230805

Reply:
Dropped `startsWith()`